src/main/scala/nl.knaw.dans.easy.bagstore/component/BagStoreComponent.scala

```diff
@@ -217,7 +217,7 @@ trait BagStoreComponent {
           staging <- if (skipStage) Try { bagDir.getParent }
                      else bagProcessing.stageBagDir(bagDir)
           path = staging.resolve(bagDir.getFileName)
-          maybeRefbags <- bagProcessing.getReferenceBags(path)
+          maybeRefbags <- bagProcessing.getReferenceBags(path, uuid)
```

Comment:
https://github.com/DANS-KNAW/easy-bag-store/blob/c6dd56901115efe56aadc71fdeab4aa29bb97ea6/src/main/scala/nl.knaw.dans.easy.bagstore/component/BagStoreComponent.scala#L210-L214

Notice that if the UUID is not known (i.e. it's a `None`), a new one is generated. Hence, there is always a UUID available. Therefore you shouldn't give an `Option[UUID]` here, but the `bagId.uuid` instead. In the code below, you are also therefore never able to reach `.getOrElse("UNKNOWN")`.

https://github.com/DANS-KNAW/easy-bag-store/blob/c6dd56901115efe56aadc71fdeab4aa29bb97ea6/src/main/scala/nl.knaw.dans.easy.bagstore/component/BagProcessingComponent.scala#L270-L275